### ISSUE TITLE
Request Builder - Built Request Affected By Initial Request

### ISSUE THREAD

**Issue by yilmaz-burak** [gavv/httpexpect#117]
Hi, 
I have a function initializing the requests
```
func InitiateClients(t *testing.T) (*httpexpect.Expect, *httpexpect.Expect) {
	managementReq := httpexpect.New(t, UserCredentials.BaseApiUrl)
	managementAuth := managementReq.Builder(func(req *httpexpect.Request) {
		req.WithBasicAuth(UserCredentials.UserEmail, UserCredentials.UserToken)
	})

	return managementReq, managementAuth
}
```

This works as expected.

When I run the requests in the following form, they behave as expected:
```
_, managementAuth := helpers.InitiateClients(t)

managementAuth.DELETE("STH").Expect().Status(<statusCode>).Body().Raw()
```
Giving the correct status 200.

However, for the following case:
```
managementReq, managementAuth := helpers.InitiateClients(t)
managementReq.DELETE("STH").Expect().Status(<statusCode>).Body().Raw()
managementAuth.DELETE("STH").Expect().Status(<statusCode>).Body().Raw()
```

Works as expected for the first request but doesn't for the second. It is as if I am calling the `managementReq`, instead of `managementAuth` one. Gives the same exact error, 401 to be exact, as in the first DELETE request.

I assume this has something to do with pointers, but I can't really understand why they wouldn't work. Docs mention some overwriting kind of stuff, but I don't think there is an overwriting case here.

Or am I completely lost, and missing something?

Thanks!

**Comment by gavv** [comment]
Hi,

I think I found the bug you're experiencing - builders slice was not copied deeply, and in some cases modifications to the copy could affect the original.

I've added a test and a fix, pushed to master. Could you check it out?

**Comment by gavv** [comment]
Tagged v2.4.0 with this fix included.

**Comment by gavv** [comment]
Added one more fix and tagged v2.4.1
